File: recyclerview_album/src/main/java/com/lt/recyclerview_album/ui/activity/AlbumDetailActivity.java
package com.lt.recyclerview_album.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.PagerSnapHelper;
import android.support.v7.widget.RecyclerView;

import com.lt.recyclerview_album.R;
import com.lt.recyclerview_album.adapter.AlbumDetailAdapter;
import com.lt.recyclerview_album.bean.AlbumBean;
import com.lt.recyclerview_album.util.ScreenUtil;

import java.util.List;

public class AlbumDetailActivity extends AppCompatActivity {
    private List<AlbumBean> mAlbumBeanList;
    private RecyclerView mRecyclerView;
    private AlbumDetailAdapter mAlbumDetailAdapter;
    private int mCurrentPosition;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_album_detail);
        bindView();
        initData();
        initView();
    }

    private void bindView() {
        mRecyclerView = findViewById(R.id.rcv_album_detail);
    }

    @SuppressWarnings("unchecked")
    private void initData() {
        mAlbumBeanList = AlbumActivity.mAlbumBeanList;
        Intent intent = getIntent();
        Bundle bundle = intent.getExtras();
        mCurrentPosition = bundle.getInt("position");
    }

    private void initView() {
        ScreenUtil.setNoStatusBar(this);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        linearLayoutManager.setOrientation(LinearLayoutManager.HORIZONTAL);
        mAlbumDetailAdapter = new AlbumDetailAdapter(R.layout.album_detail_item, mAlbumBeanList);
        mRecyclerView.setLayoutManager(linearLayoutManager);
        mRecyclerView.setAdapter(mAlbumDetailAdapter);
        mRecyclerView.scrollToPosition(mCurrentPosition);//根据bundle接收的position滑动到指定位置
        PagerSnapHelper pagerSnapHelper = new PagerSnapHelper();
        pagerSnapHelper.attachToRecyclerView(mRecyclerView);//这两句实现了ViewPage效果
    }
}

File: recyclerview_album_kotlin/src/main/java/com/lt/recyclerview_album_kotlin/ui/activity/AlbumDetailActivity.kt
package com.lt.recyclerview_album_kotlin.ui.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.PagerSnapHelper
import com.lt.recyclerview_album_kotlin.R
import com.lt.recyclerview_album_kotlin.adapter.AlbumDetailAdapter
import com.lt.recyclerview_album_kotlin.bean.AlbumBean
import com.lt.recyclerview_album_kotlin.util.ScreenUtil
import kotlinx.android.synthetic.main.activity_album_detail.*

class AlbumDetailActivity : AppCompatActivity() {
    private lateinit var mAlbumDetailAdapter: AlbumDetailAdapter
    private lateinit var mAlbumBeanList: MutableList<AlbumBean>
    private var mCurrentPosition: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_album_detail)
        initData()
        initView()
    }

    @Suppress("RECEIVER_NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
    private fun initData() {
        mAlbumBeanList = AlbumActivity.mAlbumBeanList
        val intent = intent
        val bundle = intent.extras
        mCurrentPosition = bundle.getInt("position")
    }

    private fun initView() {
        ScreenUtil.setNoStatusBar(this)
        val layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        mAlbumDetailAdapter = AlbumDetailAdapter(R.layout.album_detail_item, mAlbumBeanList)
        rcv_album_detail.layoutManager = layoutManager
        rcv_album_detail.adapter = mAlbumDetailAdapter
        rcv_album_detail.scrollToPosition(mCurrentPosition)//根据bundle接收的position滑动到指定位置
        val pagerSnapHelper = PagerSnapHelper()
        pagerSnapHelper.attachToRecyclerView(rcv_album_detail)//这两句实现了ViewPage效果
    }
}
